indexer/services/comlink/src/helpers/alchemy-helpers.ts

```
@@ -0,0 +1,291 @@
+    kernelVersion: KERNEL_V3_1,
+    eoaAddress: address as Address,
+    index: BigInt(0),
+  });
+  return kernelAddress;
+}
+
+export async function getEOAAddressFromSmartAccountAddress(
+  smartAccountAddress: Address,
+): Promise<Address> {
+  const smartAccountAddressToUse = checksumAddress(smartAccountAddress);
+  const record = await findBySmartAccountAddress(smartAccountAddressToUse);
+  if (!record || !record.evm_address) {
+    throw new Error('Failed to find a turnkey user for address');
+  }
+  return record.evm_address as Address;
+}
+
+export enum CosmosPrefix {
+  OSMO = 'osmo',
+  NEUTRON = 'neutron',
+  NOBLE = 'noble',
+}
+
+// Prefix is one of osmosis, neutron, noble. This is how we convert dydx addresses
+// to other chain addresses on cosmos. Address here is dydx address.
+export function toClientAddressWithPrefix(prefix: CosmosPrefix, address: string): string | null {
+  try {
+    const decoded = decode(address);
+    if (decoded.prefix !== 'dydx') {
+      return null;
+    }
+    return encode(prefix, decoded.words);
+  } catch (e) {
+    return null;
+  }
+}
+
+export function isSupportedEVMChainId(chainId: string): boolean {
+  return Object.keys(chains).includes(chainId);
+}
+
+export function getRPCEndpoint(chainId: string): string {
+  if (!isSupportedEVMChainId(chainId)) {
+    throw new Error(`Unsupported chainId: ${chainId}`);
+  }
+  return `${config.ZERODEV_API_BASE_URL}/${config.ZERODEV_API_KEY}/chain/${chainId}`;
+}
+
+// TODO: Verify that this function is 1000% correct. @RUI and @TYLER and @JARED
+export function getAddress(
+  chainId: string,
+  sourceAddress: string,
+  dydxAddress: string,
+): string {
+  if (isSupportedEVMChainId(chainId) || chainId === 'solana') {
+    return sourceAddress;
+  }
+  switch (chainId) {
+    case 'noble-1':
+      return toClientAddressWithPrefix(CosmosPrefix.NOBLE, dydxAddress) || '';
+    case 'osmosis-1':
+      return toClientAddressWithPrefix(CosmosPrefix.OSMO, dydxAddress) || '';
```

lets throw an error here instead of falling back onto an empty string. we dont expect this so an error is appropriate